YAML/YAML.sublime-syntax

```diff
@@ -1,97 +1,529 @@
+        7: constant.language.merge.yaml
+
+  flow-scalar-plain-out-implicit-type:
+    - match: |
+        (?x)
+        {{_type_all}}
+        \s*
+        {{_flow_scalar_end_plain_out}}
+      captures: # *implicit_type_captures
+        # Alias does not work, see https://github.com/SublimeTextIssues/Core/issues/967
+        1: constant.language.null.yaml
+        2: constant.language.boolean.yaml
+        3: constant.numeric.integer.yaml
+        4: constant.numeric.float.yaml
+        5: constant.other.timestamp.yaml
+        6: constant.language.value.yaml
+        7: constant.language.merge.yaml
+
+  flow-scalar-plain-out:
+    # http://yaml.org/spec/1.2/spec.html#style/flow/plain
+    # ns-plain(n,c) (c=flow-out, c=block-key)
+    - include: flow-scalar-plain-out-implicit-type
+    - match: '{{ns_plain_first_plain_out}}'
+      push:
+        - meta_scope: string.unquoted.plain.out.yaml
+          meta_include_prototype: false # not required to exclude, but are unnecessary
+        - match: '{{_flow_scalar_end_plain_out}}'
+          pop: true
+
+  flow-scalar-plain-in:
+    # http://yaml.org/spec/1.2/spec.html#style/flow/plain
+    # ns-plain(n,c) (c=flow-in, c=flow-key)
+    - include: flow-scalar-plain-in-implicit-type
+    - match: '{{ns_plain_first_plain_in}}'
+      push:
+        - meta_scope: string.unquoted.plain.in.yaml
+          meta_include_prototype: false # not required to exclude, but are unnecessary
+        - match: '{{_flow_scalar_end_plain_in}}'
+          pop: true
+
+  flow-sequence:
+    # http://yaml.org/spec/1.2/spec.html#style/flow/sequence
+    # c-flow-sequence(n,c)
+    - match: \[
+      scope: punctuation.definition.sequence.begin.yaml
+      push:
+        - meta_scope: meta.flow-sequence.yaml
+        - match: \]
+          scope: punctuation.definition.sequence.end.yaml
+          pop: true
+        - match: ','
+          scope: punctuation.separator.sequence.yaml
+        - include: flow-pair
+        - include: flow-node
+
+  flow-mapping:
+    - match: \{
+      scope: punctuation.definition.mapping.begin.yaml
+      push:
+        - meta_scope: meta.flow-mapping.yaml
+        - match: \}
+          scope: punctuation.definition.mapping.end.yaml
+          pop: true
+        - include: flow-pair
+
+  flow-pair:
+    - match: \?
+      scope: punctuation.definition.key-value.begin.yaml
+      push:
+        - meta_scope: meta.flow-mapping.yaml
+        - match: (?=[},\]]) # Empty mapping keys & values are allowed
+          pop: true
+        - include: flow-pair
+        - include: flow-node
+        - match: :(?=\s|$|{{c_flow_indicator}})
+          scope: punctuation.separator.mapping.key-value.yaml
+          set: flow-pair-value
+    # Attempt to match plain-in scalars and highlight as "entitiy.name.tag",
+    # if followed by a colon
+    - match: |
+        (?x)
+        (?=
+          {{ns_plain_first_plain_in}}
+          ((?!{{_flow_scalar_end_plain_in}}).)*
```

Comment:
Here you have a lookahead, containing a negative lookahead, containing a lookahead.

Proposed change: `([^ :\{\}\[\],]|\:[^ \{\}\[\],]| [^#])*`